compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -745,278 +747,278 @@ fn encode_ty<'tcx>(
+            ty::Tuple(tys) => {
+                Ty::new_tup_from_iter(self.tcx, tys.iter().map(|ty| self.fold_ty(ty)))
+            }
 
-        ty::Slice(ty0) => {
-            ty = Ty::new_slice(tcx, transform_ty(tcx, *ty0, parents, options));
-        }
+            ty::Array(ty0, len) => {
+                let len = len.eval_target_usize(self.tcx, ty::ParamEnv::reveal_all());
+                Ty::new_array(self.tcx, self.fold_ty(*ty0), len)
+            }
 
-        ty::Adt(adt_def, args) => {
-            if ty.is_c_void(tcx) {
-                ty = Ty::new_unit(tcx);
-            } else if options.contains(TransformTyOptions::GENERALIZE_REPR_C) && adt_def.repr().c()
-            {
-                ty = Ty::new_adt(tcx, *adt_def, ty::List::empty());
-            } else if adt_def.repr().transparent() && adt_def.is_struct() && !parents.contains(&ty)
-            {
-                // Don't transform repr(transparent) types with an user-defined CFI encoding to
-                // preserve the user-defined CFI encoding.
-                if let Some(_) = tcx.get_attr(adt_def.did(), sym::cfi_encoding) {
-                    return ty;
-                }
-                let variant = adt_def.non_enum_variant();
-                let param_env = tcx.param_env(variant.def_id);
-                let field = variant.fields.iter().find(|field| {
-                    let ty = tcx.type_of(field.did).instantiate_identity();
-                    let is_zst =
-                        tcx.layout_of(param_env.and(ty)).is_ok_and(|layout| layout.is_zst());
-                    !is_zst
-                });
-                if let Some(field) = field {
-                    let ty0 = tcx.type_of(field.did).instantiate(tcx, args);
-                    // Generalize any repr(transparent) user-defined type that is either a pointer
-                    // or reference, and either references itself or any other type that contains or
-                    // references itself, to avoid a reference cycle.
-
-                    // If the self reference is not through a pointer, for example, due
-                    // to using `PhantomData`, need to skip normalizing it if we hit it again.
-                    parents.push(ty);
-                    if ty0.is_any_ptr() && ty0.contains(ty) {
-                        ty = transform_ty(
-                            tcx,
-                            ty0,
-                            parents,
-                            options | TransformTyOptions::GENERALIZE_POINTERS,
-                        );
+            ty::Slice(ty0) => Ty::new_slice(self.tcx, self.fold_ty(*ty0)),
+
+            ty::Adt(adt_def, args) => {
+                if t.is_c_void(self.tcx) {
+                    Ty::new_unit(self.tcx)
+                } else if self.options.contains(TransformTyOptions::GENERALIZE_REPR_C)
+                    && adt_def.repr().c()
+                {
+                    Ty::new_adt(self.tcx, *adt_def, ty::List::empty())
```

Comment:
What the heck happened to the args lol? Pre-existing, but this seems wrong.

Reply:
We don't encode args with GENERALIZE_REPR_C for cross-language CFI, but this is already handled in encode_ty. Fixed.